### DISCUSSION TITLE
NEF mogrify

### DISCUSSION THREAD

**Discussion by artooksanen** [ImageMagick/ImageMagick#3998] [Development]
I was trying to convert hundreds of NEF files to jpeg for a time lapse, but I made an error running the command with wrong parameters on my memory card instead of a directory where I had copied just a few images for a test run. 

`magick mogrify -resize 1000x1000 *.NEF`

This command destroyed all my raw images. It turned them to very small resolution and bad looking tiff images with .NEF~ extension.  It would not have been big harm if the image quality would have stayed good, but it did not. Probably there is no way to recover the original images?

This command works just great and keeps the image quality:

`magick mogrify -format jpg -resize 1000x1000 *.NEF`

I am learning here to always write lock the memory card before playing with new commands!

ImageMagick version information:

```
Version: ImageMagick 7.1.0-4 Q16 x64 2021-07-18 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Visual C++: 192930038
Features: Cipher DPC HDRI Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
```

arto

**Comment by snibgo** [comment]
> This command destroyed all my raw images.

Oops.

There are utilities that claim to attempt to "undelete" files from memory cards. I have never used any, but they may help you.

**Comment by fmw42** [comment]
Always make a backup or use the -path option in mogrify.
